### PR TITLE
TLS connection support for redis via feature flag

### PR DESCRIPTION
**Pull request by fakeshadow**
## PR Type

Feature

## PR Checklist
Check your PR fulfills the following:

- [x] Documentation comments have been added / updated.
- [x] A changelog entry has been made for the appropriate packages.
- [x] Format code with the latest stable rustfmt

## Overview

Add `native-tls` feature for connecting to a tls enabled redis-server.

### CODE REVIEW

File: actix-redis/Cargo.toml [hunk 1]
```diff
@@ -53,6 +54,9 @@ rand = { version = "0.8.0", optional = true }
 serde = { version = "1.0.101", optional = true }
 serde_json = { version = "1.0.40", optional = true }
 
+native-tls-crate = { package = "native-tls", optional = true }
+tokio-native-tls = { version = "0.3", optional = true }
```
Comment (robjtede) [inline actix-redis/Cargo.toml:58]: tokio-native-tls re-exports it's native-tls version

**Review by robjtede — COMMENTED** [review]
the feature flag should be strictly additive, eg. a new RedisTlsActor

**Comment by fakeshadow** [comment]
redis has it's own schema for decide wether to use tls and potential secure flag.
It would be best be done with a single actor type that parse the input redis address to a uri and construct connector accordingly.
A new actor type would push this task to session module which is out the scope and compilcate the building.

The current feature flag is used to give a limit support for tls and proper builder type should be introduced later when redis authentication and possible uds support added. From there the actor would have enough information to construct according connector types.

File: actix-redis/Cargo.toml (hunk 1 above)
Comment (robjtede) [reply actix-redis/Cargo.toml:58]: actually looks like you've used it so no need for this entry

**Review by robjtede — APPROVED** [review]

**Comment by fakeshadow** [comment]
In a follow PR https://github.com/actix/actix-extras/pull/167. `actix-redis`'s `native-tls` has been relaxed and can connect to both redis and rediss prefixed redis server.

**Pull request closed without merging**